org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/participant/ResourceRelocationProcessor.xtend

```diff
@@ -90,6 +96,30 @@ class ResourceRelocationProcessor {
 		modificationApplicationMonitor.done
 		return changeConverter.change
 	}
+	
+	/**
+	 * @since 2.18
+	 */
+	protected def boolean isPersistedIndexUsageRequested(ResourceRelocationContext context) {
+		val strategies = strategyRegistry.strategies
+		val persistedIndexUsageRequested = strategies.exists[
```

Comment:
I think that the live scope is mandated if at least one strategy requires the live scope. As of now a single persisted descr request will break other strategies that rely on live scoping

Reply:
That is indeed true, and for that reason I tried it the other way round in the first place, but as @cdietrich [correctly remarked](https://github.com/eclipse/xtext-core/commit/265e4c97e3689747cf051503f5d21e870c019b06#commitcomment-33214817) that would imply a change in the default behavior.

We also discussed that compromise internally as all of the possible options of introducing that switch do have some drawbacks.

That's why I put the detailed API documentation in https://github.com/eclipse/xtext-core/blob/247a38a20e5d0db666a981dad661b89ed771da83/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/refactoring/IResourceRelocationStrategy.xtend#L32-L54, which hopefully makes the consequences of returning `true` clear (I on purpose tried to be kind of formal there).

Consequently, languages should not just override the method with a `return true` body but evaluate the information hold by the provided `ResourceRelocationContext` for making a proper decision, as long as they don't live in a closed world setup.